### ISSUE TITLE
Save state

### ISSUE THREAD

**Issue by Miloune** [wolfreak99/audio-router#4]
Hello,

Thanks for this continuation of the Audio-Router. I really appreciate this software and i have a question:

Do you think it's possible / not that hard to save the state of a process.
I mean, if i duplicate Spotify on another device output and then i close / reopen Spotify, it is possible to restore my duplicate configuration?
Another things is to save all states then close the program and restart it with the configuration saved.

What do you think about it?

Thanks for your contribution and all your work!

**Comment by wolfreak99** [comment]
Hi @Miloune , There is actually some code that was commented out reguarding a possible "save audio routing" functionality. I'm not entirely sure what it does or if it works, but I plan to look into it and see why it was commented out. If it is still buggy (the original developer commented it out due to some "technical difficulties with some programs"), I am hoping to, at the least, make two releases, one that'll include the functionality and one that will not. And depending on how it was implemented, maybe consider adding an option with a note that enabling it may cause some glitches (at least until i learn the cores of the program enough to fix it or if anything rewrite it myself)

I have not tested it yet, but if you wish to give it a shot, you can try compiling the program with "ENABLE_BOOTSTRAP" defined, which is what i've placed around some of the code that seems to fit the saved routing functionality until i can test it later on in its own worth and see how it functions. Again, it may be buggy, it might not even compile right, i haven't got around to testing the symbol.

Long story short, this is definitely something I am wanting to do.

**Comment by individuell** [comment]
looking forward on helping with this once I get back home. Thanks a lot for taking your time for this project wolfreak99!

**Comment by wolfreak99** [comment]
@individuell dude any help with this project would be so appreciated. I'm honestly really stressed with this project and since I accidentally pushed a few pull requests to the audioroiterdev's branch, this fork has caught wind and alot of people are starring it and I'm still trying to clean up some of the code and get it to a more maintainable state, yet I don't know much of what's even happening when it comes to the cores of the code. I've considered trying to port the code to c# but I doubt c# would be able to do what this code does as easily without alot of workarounds and hacks. I'm wanting to at the least *eventually* invest in another window form system, such as qT or something, to get away from all the obscure variable types and variable types with caps locks.

If you want you can send me a message on gitter.im or email, jbminor1991 at Gmail dot com. I also have a Facebook under the same email. Either way I appreciate any form of help, just make sure your work is done on a separate branch that's based off *develop*, since that's the branch that the code is updated on.

I will be getting off work somewhere between midnight and 6:30am, I'm writing this as of 7:34pm

**Comment by wolfreak99** [comment]
Right now, I'm mostly working on trying to get the code in a cleaner and more organized state, and trying to move the core audio related functionality to audio-router and not need to rely on audio-router-gui as much. Im a believer in keeping core functionality and gui functionality as separate as possible (and gui functionality use the core functionality, and not the other way around. The core functionality should be able to run with as little (if not any) gui knowledge as possible.. if that makes sense)

**Comment by Miloune** [comment]
Thanks for all of this. Please let me know if you did something about this functionality :) Have a good day.
